configs/typescript.js

```diff
@@ -0,0 +1,20 @@
+module.exports = {
+	extends: [
+		'plugin:@typescript-eslint/recommended',
+	],
+	/**
+	 * Please include a short description of the rule. For rules that downgrade or
+	 * disable errors, include a brief justification or reasoning.
+	 */
+	rules: {
```

Comment:
I know this is controversial but do we want to add `'@typescript-eslint/no-explicit-any': 'off',` here?

Reply:
Maybe even `@typescript-eslint/no-shadow`

Reply:
This is the eslint file we use for our NestJS services: https://github.com/Automattic/vip-site-details/blob/trunk/.eslintrc.js

Reply:
> I know this is controversial but

I think the answer to anything that feels controversial is to branch off into separate configs. Especially with TypeScript, we'll want to accommodate codebases that were born into TS vs codebases that are migrating. I can envision something like:

`typescript` <-- base recommended config
`typescript/migration` <-- overrides to relax things
`typescript/strict` <-- overrides to tighten things up

> This is the eslint file we use for our NestJS services

Thanks! This sounds like a great base for `typescript/strict`. I'll add it in a new PR.

Reply:
fair enough! Thanks!